src/sentry/lang/javascript/processor.py

```diff
@@ -843,6 +844,36 @@ def get_function_for_token(frame, token, previous_frame=None):
     return frame_function_name
 
 
+def fold_function_name(function_name):
+    """
+    Fold multiple consecutive occurences of the same property name into a single group, excluding the last component.
+
+    foo | foo
+    foo.foo | foo.foo
+    foo.foo.foo | {foo#2}.foo
+    bar.foo.foo | bar.foo.foo
+    bar.foo.foo.foo | bar.{foo#2}.foo
+    bar.foo.foo.onError | bar.{foo#2}.onError
+    bar.bar.bar.foo.foo.onError | {bar#3}.{foo#2}.onError
+    bar.foo.foo.bar.bar.onError | bar.{foo#2}.{bar#2}.onError
+    """
+
+    parts = function_name.split(".")
+
+    if len(parts) == 1:
+        return function_name
+
+    tail = parts.pop()
+    grouped = [list(g) for _, g in groupby(parts)]
+
+    def format_groups(p):
+        if len(p) == 1:
+            return p[0]
+        return f"\u007b{p[0]}#{len(p)}\u007d"
```

Comment:
I'm okay with that but for the record you can do `f"{{{p[0]}#{len(p)}}}"`.

Reply:
replit was not happy about this syntax for some reason, so I went with codes directly

<img width="297" alt="image" src="https://user-images.githubusercontent.com/1523305/203760171-1bf49093-5a3a-4cd3-bfb0-32137978f6ef.png">